File: python/1040.py
n = input()

(n1, n2, n3, n4) = n.split()

nota1 = float(n1)
nota2 = float(n2)
nota3 = float(n3)
nota4 = float(n4)

media = (2 * nota1 + 3 * nota2 + 4 * nota3 + 1 * nota4) / 10
aluno_em_exame = True

print("Media: %.1f" % media)
if media >= 7.0:
    print("Aluno aprovado.")
if media < 5.0:
    print("Aluno reprovado.")
if media >= 5.0 and media <= 6.9 and aluno_em_exame:
    print("Aluno em exame.")
    nota_exame = float(input())
    print("Nota do exame: %.1f" % nota_exame)
    nova_media = (nota_exame + media) / 2
    if nova_media >= 5.0:
        print("Aluno aprovado.")
    if nova_media <= 4.9:
        print("Aluno reprovado.")
    print("Media final: %.1f" % nova_media)

File: c/1040.c
#include <stdio.h>

int main()
{
  float nota1, nota2, nota3, nota4;
  float media, nota_exame, nova_media;
  int aluno_em_exame = 1;

  scanf("%f %f %f %f", &nota1, &nota2, &nota3, &nota4);

  media = (2 * nota1 + 3 * nota2 + 4 * nota3 + 1 * nota4) / 10;

  printf("Media: %.1f\n", media);
  if (media >= 7.0)
  {
    printf("Aluno aprovado.\n");
  }
  if (media < 5.0)
  {
    printf("Aluno reprovado.\n");
  }
  if (media >= 5.0 && media <= 6.9 && aluno_em_exame)
  {
    printf("Aluno em exame.\n");
    scanf("%f", &nota_exame);
    printf("Nota do exame: %.1f\n", nota_exame);
    nova_media = (nota_exame + media) / 2;
    if (nova_media >= 5.0)
    {
      printf("Aluno aprovado.\n");
    }
    if (nova_media <= 4.9)
    {
      printf("Aluno reprovado.\n");
    }
    printf("Media final: %.1f\n", nova_media);
  }

  return 0;
}
